x/nodes/keeper/validator.go

```diff
@@ -233,6 +234,11 @@ func (k Keeper) AllValidators(ctx sdk.Ctx) (validators []exported.ValidatorI) {
 	return validators
 }
 
+// ValidatorHasLessOrEqualMaxChains - Check if the validator has less or equal than the maximum chains param
+func (k Keeper) ValidatorHasLessOrEqualMaxChains(ctx sdk.Ctx, val exported.ValidatorI) bool {
```

Comment:
I'd introduce a function `GetNodeMaxChains()` that just returns `k.MaxChains(ctx)` here
and do comparison in `NewSessionNodes`.  That's more straightforward and easier to understand than having a negated condition of a function named `ValidatorHasLessOrEqualMaxChains`.  What do you think?